09/calculator_test.go

```diff
@@ -1,17 +1,64 @@
 package calculator_test
 
 import (
+	"calculator"
 	"testing"
 )
 
 func TestSubtract(t *testing.T) {
-	t.Fatal("Not implemented yet")
+	testCases := []struct {
+		a, b int
+		want int
+	}{
+		{a: 6, b: 7, want: -1},
+		{a: 9, b: 3, want: 6},
+		{a: 0, b: 0, want: 0},
+		{a: -6, b: -7, want: 1},
+		{a: -2, b: 5, want: -7},
+	}
+	for _, testCase := range testCases {
+		got := calculator.Subtract(testCase.a, testCase.b)
+		if got != testCase.want {
+			t.Errorf("Subtracted (%v,%v): Wanted %v, got %v", testCase.a, testCase.b, testCase.want, got)
+		}
+	}
 }
 
 func TestMultiply(t *testing.T) {
-	t.Fatal("Not implemented yet")
+	testCases := []struct {
+		a, b int
+		want int
+	}{
+		{a: 2, b: calculator.Multiply(2, 2), want: 8},
```

Comment:
This is a good idea, but for now let's keep the test cases really simple and not involving expressions or function calls. After all, it should make no difference to the _test_ if we write `4` instead of `calculator.Multiply(2,2)`, and it's easier to read.